common/src/main/java/io/netty5/util/AsciiString.java

```diff
@@ -1547,6 +1544,28 @@ public static boolean contentEquals(CharSequence a, CharSequence b) {
         return true;
     }
 
+    /**
+     * Create a subsequence of the given sequence.
+     * Equivalent to {@link #subSequence(int, int)} and {@link String#substring(int, int)}.
+     *
+     * @param sequence The sequence to extract from.
+     * @param start The start index, inclusive.
+     * @param end The end index, exclusive.
+     * @return The subsequence.
+     */
+    public static CharSequence substring(CharSequence sequence, int start, int end) {
+        if (sequence.getClass() == AsciiString.class) {
+            return ((AsciiString) sequence).subSequence(start, end);
+        }
+        final String str;
+        if (sequence.getClass() == String.class) {
+            str = (String) sequence;
```

Comment:
nit: I think calling `toString()` is good enough in both cases as if its a `String` it should be a NOOP.